pkg/loki/modules.go

```diff
@@ -281,18 +301,15 @@ func (t *Loki) initTableManager() error {
 	bucketClient, err := storage.NewBucketClient(t.cfg.StorageConfig.Config)
 	util.CheckFatal("initializing bucket client", err)
 
-	t.tableManager, err = chunk.NewTableManager(t.cfg.TableManager, t.cfg.SchemaConfig, maxChunkAgeForTableManager, tableClient, bucketClient)
-	if err != nil {
-		return err
+	t.tableManager, err = chunk.NewTableManager(t.cfg.TableManager, t.cfg.SchemaConfig, maxChunkAgeForTableManager, tableClient, bucketClient, prometheus.DefaultRegisterer)
+	if err == nil {
+		err = services.StartAndAwaitRunning(context.Background(), t.tableManager)
 	}
-
-	t.tableManager.Start()
-	return nil
+	return err
```

Comment:
nit: I usually prefer error case to return first.

something like

```
if err != nil {
  return err
}
return services.StartAndAwaitRunning(context.Background(), t.tableManager)
```

Reply:
I've used the same order in `initRing`, but it makes more sense there as otherwise there would need to be multiple err checks. Not so here. Fixed.